### PR TITLE
Refactor statement caching code

### PR DESCRIPTION
**Pull request by nirbhayc**
Part-1 of there work done in https://github.com/bloomberg/comdb2/pull/2310

/plugin-branch comdb2-stmt-cache-refactor

Signed-off-by: Nirbhay Choubey <nchoubey@bloomberg.net>

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/295cd1ec/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/295cd1ec/robomark.txt'>Error &#x26A0; </a>.<br>Regression testing: Success &#10003;.</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/295cd1ec/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/295cd1ec/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2657'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/cc0fefb1/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/cc0fefb1/robomark.txt'>Error &#x26A0; </a>.<br>Regression testing: Success &#10003;.</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/cc0fefb1/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/cc0fefb1/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2657'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/cc0fefb1/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/cc0fefb1/robomark.txt'>Error &#x26A0; </a>.<br>Regression testing: <b>1/396 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>backup_logfiles</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/cc0fefb1/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/cc0fefb1/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2657'>Resubmit</a></p>

File: db/sql.h [hunk 1]
```diff
@@ -1251,8 +1217,6 @@ int get_prepared_stmt(struct sqlthdstate *, struct sqlclntstate *, struct sql_st
 int get_prepared_stmt_no_lock(struct sqlthdstate *, struct sqlclntstate *, struct sql_state *, struct errstat *, int);
 int get_prepared_stmt_try_lock(struct sqlthdstate *, struct sqlclntstate *, struct sql_state *, struct errstat *, int);
 
-void put_prepared_stmt(struct sqlthdstate *, struct sqlclntstate *,
-                       struct sql_state *, int outrc);
 void sqlengine_thd_start(struct thdpool *, struct sqlthdstate *, enum thrtype);
 void sqlengine_thd_end(struct thdpool *, struct sqlthdstate *);
 void delete_stmt_caching_table(hash_t *stmt_caching_table);
```
Comment (adizaimi) [inline db/sql.h:1222]: does delete_stmt_caching_table() still exist?

Comment (nirbhayc) [reply db/sql.h]: No, it shouldn't, removed.

**Review by adizaimi — APPROVED** [review]
Looks great!
Thanks

**Pull request merged**